integration-tests/spring-rabbitmq/src/main/java/org/apache/camel/quarkus/component/spring/rabbitmq/it/SpringRabbitmqResource.java

```diff
@@ -40,51 +47,114 @@ public class SpringRabbitmqResource {
-    public static final String DIRECT_POLLING = "direct:polling";
-
-    public static final String QUERY_ROUTING_KEY = "routingKey";
-    public static final String QUERY_DIRECT = "fromDirect";
-    public static final String QUERY_EXCHANGE = "exchange";
-    public static final String QUERY_TIMEOUT = "timeout";
-
     @Inject
     ProducerTemplate producerTemplate;
 
     @Inject
     ConsumerTemplate consumerTemplate;
 
-    @Path("/getFromDirect")
+    @Path("/consume")
     @POST
     @Produces(MediaType.TEXT_PLAIN)
-    public String getFromDirect(@QueryParam(QUERY_DIRECT) String directName) {
-        return consumerTemplate.receiveBody(directName, 5000, String.class);
+    public Response consume(@QueryParam("queue") String queue,
+            @QueryParam("exchange") String exchange,
+            @QueryParam("routingKey") String routingKey) {
+        String url = "spring-rabbitmq:" + exchange + "?connectionFactory=#connectionFactory";
+        if (routingKey != null) {
+            url += "&routingKey=" + routingKey;
+        }
+        if (queue != null) {
+            url += "&queues=" + queue;
+        }
+        try {
+            return Response.ok().entity(consumerTemplate.receiveBody(url, 5000, String.class)).build();
+        } catch (RuntimeCamelException e) {
+            StringWriter sw = new StringWriter();
+            e.printStackTrace(new PrintWriter(sw));
+            return Response.status(500).entity(sw.toString()).build();
+        }
+    }
+
+    @Path("/getFromDirect")
+    @POST
+    public Response getFromDirect(String directName,
+            @QueryParam("timeout") Long timeout,
+            @QueryParam("duration") boolean duration,
+            @QueryParam("numberOfMessages") Integer numberOfMessages) {
+        try {
+            long _timeout = timeout != null ? timeout : 5000;
+            if (numberOfMessages != null) {
+                Instant start = Instant.now();
+                LinkedList<String> results = new LinkedList<>();
+                for (int i = 0; i < numberOfMessages; i++) {
+                    String msg = consumerTemplate.receiveBody(directName, _timeout, String.class);
+                    if (msg == null || msg.isEmpty()) {
+                        break;
+                    }
+                    results.add(msg);
+                }
+
+                Duration timeElapsed = Duration.between(start, Instant.now());
+                if (duration) {
+                    results.addFirst(timeElapsed.getSeconds() + "");
+                }
+
+                return Response.ok(SpringRabbitmqUtil.listToString(results)).build();
+            }
+
+            return Response.ok(consumerTemplate.receiveBody(directName, _timeout, String.class)).build();
+        } catch (RuntimeCamelException e) {
+            StringWriter sw = new StringWriter();
+            e.printStackTrace(new PrintWriter(sw));
+            return Response.status(500).entity(sw.toString()).build();
```

Comment:
Do you actually need to send the stack traces back to the client?

IIRC Camel's ErrorHandler should log the exceptions.

Reply:
I haven find a better way f doing so, the error (which I'd like to assert) is about 3 exceptions deep. Perhaps I can search for the proper Exception in the Resource and may return only relevant part of the text, so the ammount af data is not large. WDYT?

Reply:
Let me recheck the code and hopefully I can find a better solution. I agree that thias is not the ideal way of achieving so simple assert.

Reply:
Ah, sorry. I should have read the assertions more closely. Feel free to leave it as-is then.

IIRC you can set the response entity to the exception object and RestEasy will attempt to convert it (maybe to JSON). But I wouldn't spend time adjusting the code just for that.

Reply:
I found that I don't assert those stack traces anymore, so I removed that code completely.